src/evaluation/positioning.h

```diff
@@ -15,126 +17,224 @@ namespace evaluation {
 
 namespace {
 
-constexpr int32_t s_doublePawnPenalty { -10 };
-constexpr int32_t s_isolatedPawnPenalty { -10 };
-constexpr auto s_passedPawnBonus = std::to_array<int32_t>({ 0, 10, 30, 50, 75, 100, 150, 200 });
-constexpr int32_t s_semiOpenFileScore { 10 };
-constexpr int32_t s_openFileScore { 15 };
+/* TODO: tune these parameters */
+constexpr std::array<int32_t, magic_enum::enum_count<gamephase::Phases>()> s_doublePawnPenalty { -10, -10 };
+constexpr std::array<int32_t, magic_enum::enum_count<gamephase::Phases>()> s_isolatedPawnPenalty { -10, -10 };
 
-constexpr int32_t s_bishopMobilityScore { 3 };
-constexpr int32_t s_queenMobilityScore { 1 }; /* TODO: make game phase dependent */
-constexpr int32_t s_kingShieldScore { 5 };
+using PassedPawnType = std::array<int32_t, 8>;
+constexpr std::array<PassedPawnType, magic_enum::enum_count<gamephase::Phases>()> s_passedPawnBonus { {
+    { 0, 10, 30, 50, 75, 100, 150, 200 },
+    { 0, 10, 30, 50, 75, 100, 150, 200 },
+} };
+
+constexpr std::array<int32_t, magic_enum::enum_count<gamephase::Phases>()> s_semiOpenFileScore { 15, 15 };
+constexpr std::array<int32_t, magic_enum::enum_count<gamephase::Phases>()> s_openFileScore { 15, 15 };
+
+constexpr std::array<int32_t, magic_enum::enum_count<gamephase::Phases>()> s_bishopMobilityScore { 3, 3 };
+constexpr std::array<int32_t, magic_enum::enum_count<gamephase::Phases>()> s_queenMobilityScore { 1, 1 };
+constexpr std::array<int32_t, magic_enum::enum_count<gamephase::Phases>()> s_kingShieldScore { 5, 5 };
 
 }
 
 namespace position {
 
 template<Player player>
-constexpr static inline int32_t getPawnScore(const BitBoard& board, const uint64_t pawns)
+constexpr static inline gamephase::Score getPawnScore(const BitBoard& board, const uint64_t pawns)
```

Comment:
My naive profiling (move evaluation from kiwiPete position) suggests that calls to `getPawnScore` already use a similar order of magnitude of CPU time to evaluation of rook or bishop moves. 

![Screenshot from 2025-03-17 12-18-38](https://github.com/user-attachments/assets/6f411a82-34c1-481d-b56b-2751266025a8)

I think we'd need to justify the extra operations for separate mg and eg scores here with a performance gain from tweaking the parameters.

Or am I missing something?